File: cibernox/pomodoro.rb
class Pomodoro
  attr_accessor :duration, :remaining_time, :interruptions, :started, :stopped, :timer
  def initialize(time = 25)
    self.duration = self.remaining_time = time * 60 # Time in minutes
    self.stopped = true
    self.interruptions = 0
  end

  def stopped?
    stopped
  end

  def start
    self.stopped = false
    self.started = true
    work!
  end

  def work!
    self.timer = Thread.new {
      loop do
        self.remaining_time -= 5
        if self.remaining_time <= 0
          puts "End of pomodoro. Good bye!"
          exit 
        end
        sleep(2) 
      end
    }
  end

  def stop
    if self.started && !self.stopped
      self.interruptions += 1
      self.stopped = true
      Thread.kill(self.timer) if self.timer
    else
      false
    end
  end

  def resume
    self.stopped = false
    work!
  end

  def reset
    self.remaining_time = duration
    self.interruptions = 0
    Thread.kill(self.timer) if self.timer
  end

  def finished? 
    started && remaining_time <= 0
  end


end

File: icaroKol/python/pomodoroKata/pomodoro.py
__author__ = 'Yamit Cardenas'
import time
import threading


class Pomodoro(threading.Thread):

    def __init__(self, duration=25):
        super(Pomodoro, self).__init__()

        self.__setDuration(duration)
        self.started = False
        self.timeLeft = self.duration * 60
        self.finish = False
        self.interruptions = 0

    def __setDuration(self,duration):
        if(duration > 0):
            self.duration = duration
        else:
            self.duration = 25

    def startPomodoro(self):
        self.started = True
        self.start()

    def run(self):
        while self.timeLeft:
            time.sleep(1)
            self.timeLeft -= 1
        self.finish = True

    def interrupt(self):
        if(self.started):
            self.interruptions += 1


